### DISCUSSION TITLE
Lookup with Code Description not working in a detail relation

### DISCUSSION THREAD

**Discussion by stixoffire** [serenity-is/Serenity#6385] [Help] [answered]
I am using @Estrusco  code, for Lookup with Code Description but I have an issue with MasterDetail Relations and the DropDown there - it shows the Name, but the Description shows 'undefined'.

The code works fine **for a non-related table.** 
My issue is in the Master-Detail (Details Relation I do not get the description - only the name field.)
I tried placing the [Lookup attribute on the Fields in the Details Row and Form .

So I have for eample:
Master:Employee Where EmpId is PK
Detail:Training Where TrainId is PK

On the Details Tab of the Employee Form the List<Training> , I utilize the TrainId field for the drop down with TextualField of Name .

I tired putting the lookup in my EmployeeTraiining Row and tried the Form as well 
(individually and also both for grins & giggles)..

[LookupCodeDescr(typeof(Modules.Training), "Name", "Description"]
Int32? TrainId 

Any Idea what could be wrong .. is there a reason I am not getting the description field in the Detail Relationship 
I am using in the Employee [MasterDetailRelation ... blah blah] on my TrainingRow ..

**Comment by wezmag** [comment]
Try add `IncludeColumns` in your `MasterDetailRelation`
For example:
```
[MasterDetailRelation(foreignKey: "TrainingId", IncludeColumns = "NameOfDescriptionField")]
```
Above code is untested, you can give a try.

**Reply by stixoffire** [reply]
Yes the documentation states IncludeColumns is for those columns that are not real db columns for a normal select. However because I wanted to try to get the thing to work I used 
[MasterDetailRelation(foreignKey: "EmployeeId", IncludeColumns = "Fullname, Name, Description, TrainingDescription")]

I found the answer - because this is derived from LookupEditorBase the message does not appear in Intellisense , when I put in LookupScript for the base to derive from - the answer was in the intellisense.
On the field I want to come across with the name , place [LookupInclude]

XYZRow ..
[LookupInclude]
public MyDescription { get; set; }

**Comment by stixoffire** [comment, marked as answer]
I am posting this so it is known that it is answered ..


I found the answer - because this is derived from LookupEditorBase the message does not appear in Intellisense , when I put in LookupScript for the base to derive from - the answer was in the intellisense.
On the field I want to come across with the name , place [LookupInclude]

XYZRow ..
[LookupInclude]
public MyDescription { get; set; }
